File: Python3/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    
    stack = []
    for i in s:
        # 스택이 비어있지 않고, 스택의 top에 있는 문자와 현재 문자가 같다면
        if stack and stack[-1] == i:
            stack.pop() # 제거
            continue
        # 문자와 다르다면 추가
        stack.append(i)
    
    if not stack:
        answer = 1
    return answer

File: Java/프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        // 문자열의 각 문자를 순차적으로 처리
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            
            // 스택이 비어 있지 않고, 스택의 top과 현재 문자가 같으면 제거
            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop();  // 스택의 top과 같은 문자 제거
            } else {
                stack.push(c);  // 같지 않으면 현재 문자를 스택에 추가
            }
        }
        
        // 스택이 비어있으면 모든 문자가 짝지어 제거된 것
        return stack.isEmpty() ? 1 : 0;
    }
}
